app/models/miq_report/import_export.rb

```diff
@@ -36,7 +36,32 @@ def import_from_hash(report, options = nil)
       report[:db_options].deep_symbolize_keys! if report[:db_options]
 
       user = options[:user] || User.find_by_userid(options[:userid])
-      report.merge!("miq_group_id" => user.current_group_id, "user_id" => user.id)
+
+      if options[:preserve_owner]
+        userid = report.delete("userid")
+        group_description = report.delete("group_description")
+
+        report_user = userid.present? ? User.find_by_userid(userid) : User.find_by(:id => report["user_id"])
```

Comment:
@yrudman, one more minor change needed - `User.find_by_userid(userid)` should be changed to `User.find_by(:userid => userid)`. See rubocop warning

Reply:
```User.find_by_userid``` is not default Rails dynamic method, it is overwritten in ```User```  to get user in current region: 

```
def self.find_by_userid(userid)  
   in_my_region.find_by(:userid => userid) 
end
```